types/macaddress.pp

```diff
@@ -0,0 +1,2 @@
+# Matches MAC addresses
+type Simplib::Macaddress = Pattern['^[0-9a-fA-F]{1,2}:[0-9a-fA-F]{1,2}:[0-9a-fA-F]{1,2}:[0-9a-fA-F]{1,2}:[0-9a-fA-F]{1,2}:[0-9a-fA-F]{1,2}$']
```

Comment:
Couldn't each group only be matched twice, never once?

Reply:
Match of once allows for implied 0.  For example 'f' <-> '0f'.
I